db/changes_test.go

```diff
@@ -286,6 +287,58 @@ func TestDocDeletionFromChannelCoalesced(t *testing.T) {
 	printChanges(changes)
 }
 
+func TestActiveOnlyCacheUpdate(t *testing.T) {
+
+	db, testBucket := setupTestDB(t)
+	defer testBucket.Close()
+	defer tearDownTestDB(t, db)
+
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyChanges, base.KeyCache)()
+	log.Printf("Starting actual test code")
+	// Create 10 documents
+	revId := ""
+	var err error
+	for i := 1; i <= 10; i++ {
+		key := fmt.Sprintf("%s_%d", t.Name(), i)
+		body := Body{"foo": "bar"}
+		revId, _, err = db.Put(key, body)
+		require.NoError(t, err, "Couldn't create document")
+	}
+
+	// Tombstone 5 documents
+	for i := 2; i <= 6; i++ {
+		key := fmt.Sprintf("%s_%d", t.Name(), i)
+		_, err = db.DeleteDoc(key, revId)
+		require.NoError(t, err, "Couldn't delete document")
+	}
+
+	waitErr := db.WaitForPendingChanges(context.Background())
+	assert.NoError(t, waitErr)
+
+	changesOptions := ChangesOptions{
+		Since:      SequenceID{Seq: 0},
+		ActiveOnly: true,
+	}
+
+	initQueryCount, _ := base.GetExpvarAsInt("syncGateway_changeCache", "view_queries")
+
+	// Get changes with active_only=true
+	activeChanges, err := db.GetChanges(base.SetOf("*"), changesOptions)
+	require.NoError(t, err, "Error getting changes with active_only true")
+	require.Equal(t, 5, len(activeChanges))
+
+	// Ensure the test is triggering a query, and not serving from DCP-generated cache
+	postChangesQueryCount, _ := base.GetExpvarAsInt("syncGateway_changeCache", "view_queries")
+	assert.Equal(t, postChangesQueryCount, initQueryCount+1)
```

Comment:
Flip expected/actual params

```suggestion
	assert.Equal(t, initQueryCount+1, postChangesQueryCount)
```